library/include/G4CMPSolidUtils.hh

```diff
@@ -0,0 +1,59 @@
+/***********************************************************************\
+ * This software is licensed under the terms of the GNU General Public *
+ * License version 3 or later. See G4CMP/LICENSE for the full license. *
+\***********************************************************************/
+
+/// \file library/include/G4CMPSolidUtils.hh
+/// \brief Definition of the G4CMPSolidUtils class
+//
+// $Id$
+//
+// 20250424 G4CMP-465 -- Create G4CMPSolidUtils class.
+
+#ifndef G4CMPSolidUtils_hh
+#define G4CMPSolidUtils_hh 1
+
+#include "globals.hh"
+#include "G4ThreeVector.hh"
+
+class G4VSolid;
+
+class G4CMPSolidUtils {
+  public:
+
+  // Default constructor
+  G4CMPSolidUtils() : theSolid(0), verboseLevel(0) {;}
+
+  void Initialize(const G4VSolid* solid);
+
+  void SetVerboseLevel(G4int verbose=0) { verboseLevel = verbose; }
+
+  const G4VSolid* GetSolid() { return theSolid; }
+
```

Comment:
Be sure to add a comment, as well as text up in the header comment, that the position coordinate _must_ be passed in the solid's local coordinate system (but see my comment to the constructor).